### PR TITLE
added auto-translate to comments

### PR DESCRIPTION
**Pull request by quinnha**
Added support for comment language detection and auto-translation

## Description
Created `translateComments` that automatically detects the language of all comments, and translates them if they are not in English. The implementation creates and observer, and every time new comments are loaded in, it translates that batch of comments.

Also refactored translate API call to be modular for readibility, new function `callTranslateAPI` defined

## Other Notes
Comments which have multiple lines (separated by the enter key) says "(translated)" multiple times. To be fixed in another PR 👍

### CODE REVIEW

**Comment by quinnha** [comment]
Side note - this approach is pretty bad when it comes to large quantities of comments, just because it's using the translate API to check it's language, maybe there's a way where we don't have to make that call? Could be something when we get a server up and running

File: scripts/content.js [hunk 1]
```diff
@@ -1,61 +1,115 @@
 /** Google Translate API */
 
-const key = "";
-const url = `https://translation.googleapis.com/language/translate/v2?key=${key}`;
+const key = ""; // Insert Key Here
+const URL = `https://translation.googleapis.com/language/translate/v2?key=${key}`;
```
Comment (wastadtlander) [inline scripts/content.js]: rename this to translateURL

File: scripts/content.js [hunk 2]
```diff
@@ -1,61 +1,115 @@
...
+const detectURL = `https://translation.googleapis.com/language/translate/v2/detect?key=${key}`;
+const regex_emoji =
```
Comment (wastadtlander) [inline scripts/content.js]: camel case bruhpls

File: scripts/content.js [hunk 3]
```diff
@@ -1,61 +1,115 @@
...
+        });
+    } else {
+      resolve(null);
+    }
+  });
+}
+
+function translateComments(mutationsList, observer) {
+  for (const mutation of mutationsList) {
+    if (mutation.type === "childList") {
```
Comment (wastadtlander) [inline scripts/content.js]: can just use `==`

File: scripts/content.js [hunk 4]
```diff
@@ -1,61 +1,115 @@
...
+      mutation.addedNodes.forEach((node) => {
+        if (node.nodeName.toLowerCase() === "ytd-comment-renderer") {
```
Comment (wastadtlander) [inline scripts/content.js]: just use `==`

File: scripts/content.js [hunk 5]
```diff
@@ -1,61 +1,115 @@
...
+          let bodyElement = node.querySelectorAll(
+            "#body #main #comment-content #expander #content #content-text .style-scope.yt-formatted-string"
+          );
+
+          if (bodyElement) {
+            bodyElement.forEach((element) => {
+              const spellcheckValue = element.getAttribute("spellcheck");
+              if (
```
Comment (wastadtlander) [inline scripts/content.js:60]: can u inline so like 
```
if (condition &&
    nextCondition...
    ...)
```

File: scripts/content.js [hunk 6]
```diff
@@ -1,61 +1,115 @@
...
+                element.textContent.trim() !== "" &&
```
Comment (wastadtlander) [inline scripts/content.js]: !=

File: scripts/content.js [hunk 7]
```diff
@@ -1,61 +1,115 @@
...
+                spellcheckValue != "false" &&
```
Comment (wastadtlander) [inline scripts/content.js:62]: this hurts me but this is fine

File: scripts/content.js [hunk 8]
```diff
@@ -1,61 +1,115 @@
...
-    .then(data => {
-      const translatedComment = data.data.translations[0].translatedText;
-      console.log(translatedComment);
-      comment.textContent = translatedComment;
+
+  callTranslateAPI(comment)
+    .then((translatedText) => {
+      if (translatedText) {
+        comment.textContent = translatedText;
+        console.log("Translated text:", translatedText);
```
Comment (wastadtlander) [inline scripts/content.js:112]: is this causing double logs? line 75 is also logging after calling callTranslateAPI

**Review by wastadtlander — CHANGES_REQUESTED** [review]
could add doc comments maybe? i can do that tho also

File: scripts/content.js (hunk 2 above)
Comment (quinnha) [reply scripts/content.js]: bruv

File: scripts/content.js (hunk 5 above)
Comment (quinnha) [reply scripts/content.js:60]: my autolinter literally wont let me do it

File: scripts/content.js (hunk 8 above)
Comment (quinnha) [reply scripts/content.js:112]: callTranslateAPI doesnt log anything except the error, edited the error to avoid double logging

**Review by quinnha — COMMENTED** [review]
done

File: scripts/content.js [hunk 9]
```diff
@@ -102,12 +112,12 @@ commentWrapper.addEventListener("click", (event) => {
         console.log("Translated text:", translatedText);
```
Comment (wastadtlander) [inline scripts/content.js:112]: doesnt this still have print to console twice with line 84?

File: scripts/content.js [hunk 10]
```diff
@@ -1,61 +1,125 @@
...
-      comment.textContent = translatedComment;
-    })
-    .catch(error => {
-      console.log("Error: ", error);
+
+  callTranslateAPI(comment)
+    .then((translatedText) => {
+      if (translatedText) {
+        comment.textContent = translatedText;
+        console.log("Translated text:", translatedText);
```
Comment (wastadtlander) [inline scripts/content.js:112]: doesnt this still have print to console twice with line 84?

Comment (quinnha) [reply scripts/content.js:112]: this will console log on click vs the other console logs on load, thought we were separating the functionalities

**Review by wastadtlander — APPROVED** [review]
ho

**Pull request merged**